File: Models/Authentication/Password/UpdatePassword.cs
namespace aliment_backend.Models.Authentication.Password
{
    /// <summary>
    /// Représente les données nécessaires pour mettre à jour le mot de passe d'un utilisateur.
    /// </summary>
    public class UpdatePassword
    {
        /// <summary>
        /// Obtient ou définit les informations sur le nouveau mot de passe.
        /// </summary>
        public CreatePassword? Password { get; set; }

        /// <summary>
        /// Obtient ou définit le nom d'utilisateur de l'utilisateur dont le mot de passe doit être mis à jour.
        /// </summary>
        public string? Username { get; set; }
    }
}

File: src/app/features/auth/_models/updatePassword.ts
import { CreatePassword } from "./createPassword";

/**
 * L'interface `UpdatePassword` est utilisée pour représenter les informations nécessaires à la mise à jour du mot de passe d'un utilisateur.
 * Elle inclut les données du nouveau mot de passe (CreatePassword) ainsi que le nom d'utilisateur associé.
 * Elle est généralement utilisée dans le cadre d'une fonctionnalité permettant à un utilisateur de modifier son mot de passe.
 */
export interface UpdatePassword {
  /**
   * Nouveau mot de passe à définir lors de la mise à jour.
   * @type {CreatePassword}
   */
  password: CreatePassword;

  /**
   * Nom d'utilisateur associé à la mise à jour du mot de passe.
   * @type {string}
   */
  userName: string;
}
